packages/wallet-types/src/types.ts

```diff
@@ -1,8 +1,392 @@
+   */
+  onAccountChanged: EventHandler<(account: Account) => void | Promise<void>>;
+
+  /**
+   * The event handler for successful transaction events
+   */
+  onTxSuccessful: EventHandler<(tx: TxResponse) => void | Promise<void>>;
+
+  /**
+   * The event handler for unsuccessful transaction events
+   */
+  onTxFailed: EventHandler<(tx: TxsResponse) => void | Promise<void>>;
+}
+
+/**
+ * The signing API is used be dApp developers to be able to sign transactions and other interactions with the network
+ *
+ * By design, we expect most users to build and sign transactions by using CosmJS. By comparison with the original
+ * Keplr we do not allow other signing messages
+ */
+export interface SigningApi {
+  /**
+   * Signs arbitrary data
+   *
+   * This API can be very useful when authenticating users by their wallet address
+   *
+   * @param chainId The target chain id
+   * @param signer The target signer
+   * @param data The data that should have been signed
+   * @returns A signature object that can be verified
+   */
+  signArbitrary(
+    chainId: string,
+    signer: string,
+    data: string | Uint8Array
+  ): Promise<Signature>;
+
+  /**
+   * Verifies a signature made via the `signArbitrary` API
+   *
+   * @param chainId The target chain id
+   * @param signer The target signer
+   * @param data The data that should have been signed
+   * @param signature The signature to verify
+   * @returns True if the signature is verified, otherwise false.
+   */
+  verifyArbitrary(
+    chainId: string,
+    signer: string,
+    data: string | Uint8Array,
+    signature: Signature
+  ): Promise<boolean>;
+
+  /**
+   * Get a signer capable of either Direct or Amino signing methods.
+   *
+   * In the case that the underlying signer can support both, the direct signer will be used by default (since it is
+   * newer).
+   *
+   * @param chainId The targeted chain id
+   * @returns An CosmJS compatible signer
+   */
+  getOfflineSigner(
+    chainId: string
+  ): Promise<OfflineDirectSigner | OfflineSigner>;
+
+  /**
+   * Get a signer that supports Direct only signing methods
+   *
+   * @param chainId The target chain id
+   * @returns A CosmJS compatible signer
+   */
+  getOfflineDirectSigner(chainId: string): Promise<OfflineDirectSigner>;
+
+  /**
+   * Get a signer that supports Amino only signing methods.
+   *
+   * @param chainId The target chain id
+   * @returns A CosmJS compatible signer
+   */
+  getOfflineAminoSigner(chainId: string): Promise<OfflineSigner>;
+
+  // TODO(EJF): Need to add methods for BLS aggregated signatures
+}
+
+/**
+ * The status of the wallet.
+ */
+export type WalletStatus = "locked" | "unlocked";
+
+/**
+ * The WalletAPI
+ *
+ * The main goal of the wallet api is to provide a good interface for dApp builders in order to build compelling
+ * interfaces.
+ *
+ * The interface is designed to be extended over time.
+ */
+export interface WalletApi {
```

Comment:
Yeah the "connect" functionality is sort of the thing that I have done away with in the api. Essentially the idea is that the wallet is either connected / pointing to a network or it isn't (because the extension is locked).

This is reflected in the networks API:

https://github.com/fetchai/fetch-wallet/blob/feature/fetch-wallet-api/packages/wallet-types/src/types.ts#L46-L76

I agree with your second point, I had imagined actually that a user can call [`listNetworks`](https://github.com/fetchai/fetch-wallet/blob/feature/fetch-wallet-api/packages/wallet-types/src/types.ts#L75) or [`currentNetwork`](https://github.com/fetchai/fetch-wallet/blob/feature/fetch-wallet-api/packages/wallet-types/src/types.ts#L52). In the case where the settings here were not quite right - for example gas fees or some other configuration then the dApp can call [`switchToNetwork`](https://github.com/fetchai/fetch-wallet/blob/feature/fetch-wallet-api/packages/wallet-types/src/types.ts#L60). The main idea here being to give the dApp author quite alot of control over the network configuration. 

Obviously from a product perspective the user would have to approve these changes and so ultimately have control. What I wanted to avoid is the `connectNetwork("foo-net")` have it fail and then have to call `experimentalSuggestChain` this always felt a little clumsy to me